core/src/main/java/hudson/model/WorkspaceCleanupThread.java

```diff
@@ -139,6 +139,14 @@ private boolean shouldBeDeleted(@Nonnull TopLevelItem item, FilePath dir, @Nonnu
             }
         }
 
+        // TODO this may only check the last build in fact:
```

Comment:
This is right. AbstractByuid#isBuilding() API checks only the last job, so it won't behave reliably for parallel AbstractBuild launches.